<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.347">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Zifo RnD Solutions">
<meta name="dcterms.date" content="2022-05-27">

<title>Single Cell Gene Expression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>

<script src="droplet-processing_files/libs/clipboard/clipboard.min.js"></script>
<script src="droplet-processing_files/libs/quarto-html/quarto.js"></script>
<script src="droplet-processing_files/libs/quarto-html/popper.min.js"></script>
<script src="droplet-processing_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="droplet-processing_files/libs/quarto-html/anchor.min.js"></script>
<link href="droplet-processing_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="droplet-processing_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="droplet-processing_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="droplet-processing_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="droplet-processing_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Single Cell Gene Expression</h1>
<p class="subtitle lead">Droplet processing</p>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Zifo RnD Solutions </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 27, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<!-- Include big data processing -->
<!-- Nearest neighbour detection -->
<div class="cell">

</div>
<!-- Singular value decomposition -->
<div class="cell">

</div>
<!-- Parallel evaluation -->
<div class="cell">

</div>
<!-- Load Bioconductor package(s) -->
<div class="cell">

</div>
<!-- Load CRAN package(s) -->
<div class="cell">

</div>
<!-- Source helper function(s) -->
<div class="cell">

</div>
<!-- Set custom theme -->
<div class="cell">

</div>
<!-- Read experiment object(s) -->
<div class="cell">

</div>
<section id="barcode-ranks" class="level2">
<h2 class="anchored" data-anchor-id="barcode-ranks">Barcode ranks</h2>
<!-- Compute barcode rank statistics -->
<div class="cell">

</div>
<!-- Plot barcode rank statistics -->
<div class="cell">
<div class="cell-output-display">
<p><img src="droplet-processing_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<!-- Add barcode rank statistics to column data -->
<div class="cell">

</div>
</section>
<section id="empty-droplets" class="level2">
<h2 class="anchored" data-anchor-id="empty-droplets">Empty droplets</h2>
<!-- Identify empty droplets -->
<div class="cell">

</div>
<!-- Identify empty droplets with ambient testing -->
<div class="cell">

</div>
<p>The table below shows the number of droplets containing cells identified by the emptyDrops algorithm using a 0.001 FDR threshold.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Sample</th>
<th style="text-align: right;">Cell</th>
<th style="text-align: right;">Empty</th>
<th style="text-align: right;">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">frozen_pbmc_donor_a</td>
<td style="text-align: left;">frozen_pbmc_donor_a</td>
<td style="text-align: right;">3112</td>
<td style="text-align: right;">734168</td>
<td style="text-align: right;">737280</td>
</tr>
<tr class="even">
<td style="text-align: left;">frozen_pbmc_donor_b</td>
<td style="text-align: left;">frozen_pbmc_donor_b</td>
<td style="text-align: right;">8377</td>
<td style="text-align: right;">728903</td>
<td style="text-align: right;">737280</td>
</tr>
<tr class="odd">
<td style="text-align: left;">frozen_pbmc_donor_c</td>
<td style="text-align: left;">frozen_pbmc_donor_c</td>
<td style="text-align: right;">10017</td>
<td style="text-align: right;">1464543</td>
<td style="text-align: right;">1474560</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The table below shows how many droplets were limited by the number of iterations used for the Monte Carlo p-value calculations by the emptyDrops algorithm.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Sample</th>
<th style="text-align: right;">Limited</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">frozen_pbmc_donor_a</td>
<td style="text-align: left;">frozen_pbmc_donor_a</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">frozen_pbmc_donor_b</td>
<td style="text-align: left;">frozen_pbmc_donor_b</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">frozen_pbmc_donor_c</td>
<td style="text-align: left;">frozen_pbmc_donor_c</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">LogProb</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Rank</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">PValue</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<!-- Plot droplet log probability -->
<div class="cell">
<div class="cell-output-display">
<p><img src="droplet-processing_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<!-- Plot droplet rank -->
<div class="cell">
<div class="cell-output-display">
<p><img src="droplet-processing_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<!-- Plot ambient P value -->
<div class="cell">
<div class="cell-output-display">
<p><img src="droplet-processing_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</div>
<!-- Add empty droplet statistics to column data -->
<div class="cell">

</div>
<!-- Filter droplets by FDR threshold -->
<div class="cell">

</div>
<!-- Save experiment object(s) -->
<div class="cell">

</div>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<section id="parameters" class="level3">
<h3 class="anchored" data-anchor-id="parameters">Parameters</h3>
<p>This table shows the parameters used to process the data.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 6%">
<col style="width: 3%">
<col style="width: 90%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Parameter</th>
<th style="text-align: left;">Value</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">lower</td>
<td style="text-align: left;">100</td>
<td style="text-align: left;">A numeric scalar specifying the lower bound on the total UMI count, at or below which all barcodes are assumed to correspond to empty droplets.</td>
</tr>
<tr class="even">
<td style="text-align: left;">niters</td>
<td style="text-align: left;">10000</td>
<td style="text-align: left;">An integer scalar specifying the number of iterations to use for the Monte Carlo p-value calculations.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">fdr</td>
<td style="text-align: left;">0.001</td>
<td style="text-align: left;">Barcode FDR threshold</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="output-files" class="level3">
<h3 class="anchored" data-anchor-id="output-files">Output files</h3>
<p>This table provides links to the processed output files.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">File</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">droplet-processing.rds</td>
<td style="text-align: left;">A list of SingleCellExperiment objects.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="session-information" class="level3">
<h3 class="anchored" data-anchor-id="session-information">Session information</h3>
<p>This output lists version information about R, the OS and attached or loaded packages.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.2.0                ggplot2_3.3.6              
 [3] patchwork_1.1.1             DropletUtils_1.12.3        
 [5] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
 [7] Biobase_2.52.0              GenomicRanges_1.44.0       
 [9] GenomeInfoDb_1.28.4         IRanges_2.26.0             
[11] S4Vectors_0.30.2            BiocGenerics_0.38.0        
[13] MatrixGenerics_1.4.3        matrixStats_0.62.0         
[15] BiocParallel_1.26.2         BiocSingular_1.8.1         
[17] BiocNeighbors_1.10.0       

loaded via a namespace (and not attached):
 [1] edgeR_3.34.1              jsonlite_1.8.0           
 [3] DelayedMatrixStats_1.14.3 scuttle_1.2.1            
 [5] R.utils_2.11.0            assertthat_0.2.1         
 [7] highr_0.9                 dqrng_0.3.0              
 [9] GenomeInfoDbData_1.2.6    yaml_2.3.5               
[11] pillar_1.7.0              lattice_0.20-45          
[13] glue_1.6.2                limma_3.48.3             
[15] beachmat_2.8.1            digest_0.6.29            
[17] XVector_0.32.0            colorspace_2.0-3         
[19] htmltools_0.5.2           Matrix_1.4-1             
[21] R.oo_1.24.0               pkgconfig_2.0.3          
[23] zlibbioc_1.38.0           purrr_0.3.4              
[25] ScaledMatrix_1.0.0        HDF5Array_1.20.0         
[27] tibble_3.1.7              farver_2.1.0             
[29] generics_0.1.2            ellipsis_0.3.2           
[31] withr_2.5.0               cli_3.3.0                
[33] magrittr_2.0.3            crayon_1.5.1             
[35] evaluate_0.15             R.methodsS3_1.8.1        
[37] fansi_1.0.3               tools_4.1.1              
[39] lifecycle_1.0.1           stringr_1.4.0            
[41] Rhdf5lib_1.14.2           munsell_0.5.0            
[43] locfit_1.5-9.5            DelayedArray_0.18.0      
[45] irlba_2.3.5               compiler_4.1.1           
[47] rsvd_1.0.5                rlang_1.0.2              
[49] rhdf5_2.36.0              grid_4.1.1               
[51] RCurl_1.98-1.6            rhdf5filters_1.4.0       
[53] rstudioapi_0.13           htmlwidgets_1.5.4        
[55] labeling_0.4.2            bitops_1.0-7             
[57] rmarkdown_2.14            gtable_0.3.0             
[59] DBI_1.1.2                 R6_2.5.1                 
[61] knitr_1.39                dplyr_1.0.9              
[63] fastmap_1.1.0             utf8_1.2.2               
[65] stringi_1.7.6             Rcpp_1.0.8.3             
[67] vctrs_0.4.1               tidyselect_1.1.2         
[69] xfun_0.31                 sparseMatrixStats_1.4.2  </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>