<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.347">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Zifo RnD Solutions">
<meta name="dcterms.date" content="2022-06-18">

<title>Single Cell Gene Expression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>

<script src="feature-selection_files/libs/clipboard/clipboard.min.js"></script>
<script src="feature-selection_files/libs/quarto-html/quarto.js"></script>
<script src="feature-selection_files/libs/quarto-html/popper.min.js"></script>
<script src="feature-selection_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="feature-selection_files/libs/quarto-html/anchor.min.js"></script>
<link href="feature-selection_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="feature-selection_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="feature-selection_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="feature-selection_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="feature-selection_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Single Cell Gene Expression</h1>
<p class="subtitle lead">Feature selection</p>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Zifo RnD Solutions </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 18, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview</h2>
<p>This report contains results from the feature selection step of the Single Cell Gene Expression workflow.</p>
<blockquote class="blockquote">
<p>“Feature selection identifies genes with the strongest biological signal relative to the technical noise. By restricting downstream analysis to the most informative genes, the effect of dimensionality is diminished, noise is reduced and the analysis is simplified.”</p>
</blockquote>
<!-- Include big data parameters -->
<!-- Nearest neighbour detection -->
<div class="cell">

</div>
<!-- Singular value decomposition -->
<div class="cell">

</div>
<!-- Parallel evaluation -->
<div class="cell">

</div>
<!-- Load Bioconductor packages -->
<div class="cell">

</div>
<!-- Source custom functions -->
<div class="cell">

</div>
<!-- Read experiment objects -->
<div class="cell">

</div>
<!-- Model the per-gene variance with Poisson noise -->
<div class="cell">

</div>
<!-- Add per-gene variance statistics to row data -->
<div class="cell">

</div>
<!-- Identify highly-variable genes -->
<div class="cell">

</div>
<!-- Add HVG to row data -->
<div class="cell">

</div>
<!-- Save experiment objects -->
<div class="cell">

</div>
</section>
<section id="highly-variable-genes" class="level2">
<h2 class="anchored" data-anchor-id="highly-variable-genes">Highly variable genes</h2>
<p>Model the variance of the log-expression profiles for each gene, decomposing it into technical and biological components based on a mean-variance trend corresponding to Poisson noise.</p>
<p>Figure 1: Variance of normalized log-expression values for each gene in the dataset, plotted against the mean log-expression. The blue line represents represents the mean-variance relationship corresponding to Poisson noise.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="feature-selection_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="fig-upset" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="feature-selection_files/figure-html/fig-upset-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption aria-hidden="true" class="figure-caption">Figure 1: Highly-variable genes shared across samples</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<section id="output" class="level3">
<h3 class="anchored" data-anchor-id="output">Output</h3>
<p>This table describes the output files produced by this document.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">File</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">output/feature-selection.rds</td>
<td style="text-align: left;">A SingleCellExperiment object.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="parameters" class="level3">
<h3 class="anchored" data-anchor-id="parameters">Parameters</h3>
<p>This table describes parameters used and set in this document.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 13%">
<col style="width: 7%">
<col style="width: 78%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Parameter</th>
<th style="text-align: left;">Value</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">n</td>
<td style="text-align: left;">2500</td>
<td style="text-align: left;">Integer scalar specifying the number of top HVGs to report.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="version" class="level3">
<h3 class="anchored" data-anchor-id="version">Version</h3>
<p>This output shows the version information about R, the OS and attached or loaded packages.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.2.0                ggrepel_0.9.1              
 [3] ggplot2_3.3.6               scran_1.20.1               
 [5] scuttle_1.2.1               SingleCellExperiment_1.14.1
 [7] SummarizedExperiment_1.22.0 Biobase_2.52.0             
 [9] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[11] IRanges_2.26.0              S4Vectors_0.30.2           
[13] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[15] matrixStats_0.62.0          BiocParallel_1.26.2        
[17] BiocSingular_1.8.1          BiocNeighbors_1.10.0       

loaded via a namespace (and not attached):
 [1] bitops_1.0-7              UpSetR_1.4.0             
 [3] tools_4.1.1               utf8_1.2.2               
 [5] R6_2.5.1                  irlba_2.3.5              
 [7] DBI_1.1.2                 colorspace_2.0-3         
 [9] withr_2.5.0               tidyselect_1.1.2         
[11] gridExtra_2.3             compiler_4.1.1           
[13] cli_3.3.0                 DelayedArray_0.18.0      
[15] labeling_0.4.2            stringr_1.4.0            
[17] digest_0.6.29             rmarkdown_2.14           
[19] XVector_0.32.0            pkgconfig_2.0.3          
[21] htmltools_0.5.2           sparseMatrixStats_1.4.2  
[23] highr_0.9                 fastmap_1.1.0            
[25] limma_3.48.3              htmlwidgets_1.5.4        
[27] rlang_1.0.2               rstudioapi_0.13          
[29] DelayedMatrixStats_1.14.3 generics_0.1.2           
[31] farver_2.1.0              jsonlite_1.8.0           
[33] dplyr_1.0.9               RCurl_1.98-1.6           
[35] magrittr_2.0.3            GenomeInfoDbData_1.2.6   
[37] patchwork_1.1.1           Matrix_1.4-1             
[39] Rcpp_1.0.8.3              munsell_0.5.0            
[41] fansi_1.0.3               lifecycle_1.0.1          
[43] stringi_1.7.6             yaml_2.3.5               
[45] edgeR_3.34.1              zlibbioc_1.38.0          
[47] plyr_1.8.7                grid_4.1.1               
[49] dqrng_0.3.0               crayon_1.5.1             
[51] lattice_0.20-45           beachmat_2.8.1           
[53] locfit_1.5-9.5            metapod_1.0.0            
[55] knitr_1.39                pillar_1.7.0             
[57] igraph_1.3.1              ScaledMatrix_1.0.0       
[59] glue_1.6.2                evaluate_0.15            
[61] vctrs_0.4.1               gtable_0.3.0             
[63] purrr_0.3.4               assertthat_0.2.1         
[65] xfun_0.31                 rsvd_1.0.5               
[67] tibble_3.1.7              cluster_2.1.3            
[69] bluster_1.2.1             statmod_1.4.36           
[71] ellipsis_0.3.2           </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>