<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-0.9.347">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Zifo RnD Solutions">
<meta name="dcterms.date" content="2022-06-18">

<title>Single Cell Gene Expression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>

<script src="normalization_files/libs/clipboard/clipboard.min.js"></script>
<script src="normalization_files/libs/quarto-html/quarto.js"></script>
<script src="normalization_files/libs/quarto-html/popper.min.js"></script>
<script src="normalization_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="normalization_files/libs/quarto-html/anchor.min.js"></script>
<link href="normalization_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link id="quarto-text-highlighting-styles" href="normalization_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet">
<script src="normalization_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="normalization_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="normalization_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Single Cell Gene Expression</h1>
<p class="subtitle lead">Normalization</p>
</div>





<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Zifo RnD Solutions </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 18, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview</h2>
<!-- Include big data parameters -->
<!-- Nearest neighbour detection -->
<div class="cell">

</div>
<!-- Singular value decomposition -->
<div class="cell">

</div>
<!-- Parallel evaluation -->
<div class="cell">

</div>
<!-- Load Bioconductor package(s) -->
<div class="cell">

</div>
<!-- Source utility function(s) -->
<div class="cell">

</div>
<!-- Read experiment object(s) -->
<div class="cell">

</div>
<!-- Quick clustering of cells -->
<div class="cell">

</div>
<!-- Normalization by deconvolution -->
<div class="cell">

</div>
<!-- Compute log-normalized expression values -->
<div class="cell">

</div>
<!-- Save experiment object(s) -->
<div class="cell">

</div>
<div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">frozen_pbmc_donor_a</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">frozen_pbmc_donor_b</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">frozen_pbmc_donor_c</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell-output-display">
<p><img src="normalization_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell-output-display">
<p><img src="normalization_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="cell-output-display">
<p><img src="normalization_files/figure-html/unnamed-chunk-9-3.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</div>
</section>
<section id="summary" class="level2">
<h2 class="anchored" data-anchor-id="summary">Summary</h2>
<section id="output" class="level3">
<h3 class="anchored" data-anchor-id="output">Output</h3>
<p>This table describes the output files produced by this document.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">File</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">output/normalization.rds</td>
<td style="text-align: left;">A SingleCellExperiment object.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="parameters" class="level3">
<h3 class="anchored" data-anchor-id="parameters">Parameters</h3>
<p>This table describes parameters used and set in this document.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 8%">
<col style="width: 4%">
<col style="width: 86%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Parameter</th>
<th style="text-align: left;">Value</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">downsample</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">Logical scalar indicating whether downsampling should be performed prior to scaling and log-transformation.</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="version" class="level3">
<h3 class="anchored" data-anchor-id="version">Version</h3>
<p>This output describe the version information about R, the OS and attached or loaded packages.</p>
<!-- Print session information -->
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.2.0                ggplot2_3.3.6              
 [3] scran_1.20.1                scuttle_1.2.1              
 [5] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
 [7] Biobase_2.52.0              GenomicRanges_1.44.0       
 [9] GenomeInfoDb_1.28.4         IRanges_2.26.0             
[11] S4Vectors_0.30.2            BiocGenerics_0.38.0        
[13] MatrixGenerics_1.4.3        matrixStats_0.62.0         
[15] BiocParallel_1.26.2         BiocSingular_1.8.1         
[17] BiocNeighbors_1.10.0       

loaded via a namespace (and not attached):
 [1] edgeR_3.34.1              jsonlite_1.8.0           
 [3] DelayedMatrixStats_1.14.3 assertthat_0.2.1         
 [5] statmod_1.4.36            highr_0.9                
 [7] dqrng_0.3.0               GenomeInfoDbData_1.2.6   
 [9] yaml_2.3.5                pillar_1.7.0             
[11] lattice_0.20-45           glue_1.6.2               
[13] limma_3.48.3              beachmat_2.8.1           
[15] digest_0.6.29             XVector_0.32.0           
[17] colorspace_2.0-3          htmltools_0.5.2          
[19] Matrix_1.4-1              pkgconfig_2.0.3          
[21] zlibbioc_1.38.0           purrr_0.3.4              
[23] ScaledMatrix_1.0.0        tibble_3.1.7             
[25] farver_2.1.0              generics_0.1.2           
[27] ellipsis_0.3.2            withr_2.5.0              
[29] cli_3.3.0                 magrittr_2.0.3           
[31] crayon_1.5.1              evaluate_0.15            
[33] fansi_1.0.3               bluster_1.2.1            
[35] tools_4.1.1               lifecycle_1.0.1          
[37] stringr_1.4.0             munsell_0.5.0            
[39] locfit_1.5-9.5            cluster_2.1.3            
[41] DelayedArray_0.18.0       irlba_2.3.5              
[43] compiler_4.1.1            rsvd_1.0.5               
[45] rlang_1.0.2               grid_4.1.1               
[47] RCurl_1.98-1.6            rstudioapi_0.13          
[49] htmlwidgets_1.5.4         igraph_1.3.1             
[51] labeling_0.4.2            bitops_1.0-7             
[53] rmarkdown_2.14            gtable_0.3.0             
[55] DBI_1.1.2                 R6_2.5.1                 
[57] knitr_1.39                dplyr_1.0.9              
[59] fastmap_1.1.0             utf8_1.2.2               
[61] metapod_1.0.0             stringi_1.7.6            
[63] Rcpp_1.0.8.3              vctrs_0.4.1              
[65] tidyselect_1.1.2          xfun_0.31                
[67] sparseMatrixStats_1.4.2  </code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>