---
title: "Single Cell Gene Expression"
subtitle: "Data loading"
author: "Zifo RnD Solutions"
date: "`r format(Sys.time(), '%d %B, %Y')`"
params:
  samples: "data/samples.tsv"
  annotation: "EnsDb.Hsapiens.v86"
execute:
  message: false
  warning: false
  echo: false
---

## Introduction

This report contains the sample and feature metadata used to load and annotate the Cell Ranger output directories for 10x Genomics data.

<!-- Load Bioconductor package(s) -->

```{r}
library(DropletUtils)

library(batchelor)

library(scuttle)
```

<!-- Load CRAN package(s) -->

```{r}
library(DT)

library(skimr)
```

<!-- Load gene annotation object -->

```{r}
library(params$annotation, character.only = TRUE)

ann <- eval(parse(text = params$annotation))
```

## Sample metadata

:::{.callout-important}
Please check the sample annotation table is correct.
:::

<!-- Read sample annotation table -->

```{r}
tab <- read.delim(params$samples)
```

<!-- Display sample annotation table -->

```{r}
#| label: fig-samples
#| fig-cap: Sample annotation table.

datatable(tab)
```

:::{.callout-important}
Please check the sample summary statistics are appropriate.
:::

<!-- Summarize sample annotation table -->

```{r}
#| label: fig-summary
#| fig-cap: Sample summary table

skim(tab, .data_name = "Sample annotation table")
```

<!-- Load data from experiment(s) -->

```{r}
sce <- read10xCounts(tab$sample, sample.names = tab$sample.name)
```

<!-- Display droplet count table -->

```{r}
#| label: fig-cells
#| fig-cap: Droplet count table.

tab <- table(sce$Sample)

tab <- as.data.frame(tab)

datatable(tab, colnames = c("Sample" = "Var1", "Count" = "Freq"))
```

## Feature metadata

:::{.callout-important}
Please check the feature data is appropriate for the experiment.
:::

<!-- Make feature names unique -->

```{r}
rownames(sce) <- uniquifyFeatureNames(
  ID = rowData(sce)$ID,
  names = rowData(sce)$Symbol
)
```

<!-- Get the chromosome name -->

```{r}
rowData(sce)$Chrom <- mapIds(
  x = ann,
  keys = rowData(sce)$ID,
  column = "SEQNAME",
  keytype = "GENEID"
)
```

<!-- Display feature annotation metadata -->

```{r}
tab <- c(
  Package  = params$annotation,
  Genome   = unique(genome(ann)),
  Organism = organism(ann),
  Provider = seqlevelsStyle(ann)
)

tab <- as.data.frame(tab)

datatable(
  data = tab,
  options = list(
    dom = "t",
    ordering = FALSE,
    searching = FALSE
  ),
  colnames = ""
)
```

<!-- Divide experiment object into batches -->

```{r}
sce <- divideIntoBatches(sce, sce$Sample)$batches
```

<!-- Save experiment object(s) -->

```{r}
saveRDS(sce, file = "output/data-loading.rds")
```
